src/neptune/cmd.d

```diff
@@ -0,0 +1,208 @@
+/*******************************************************************************
+
+    Command line interaction helper
+
+    Copyright:
+        Copyright (c) 2017 Sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module neptune.cmd;
+
+/// Exception to be thrown when the exit code is unexpected
+class ExitCodeException : Exception
+{
+    this ( string msg, string file = __FILE__, int line = __LINE__ )
+    {
+        super(msg, file, line);
+    }
+}
+
+/*******************************************************************************
+
+    Runs a command and returns the output
+
+    Params:
+        command = command to run
+
+    Returns:
+        output of the command
+
+*******************************************************************************/
+
+string cmd ( string command )
+{
+    import std.process : executeShell;
+    import std.string : strip;
+
+    auto c = executeShell(command);
+
+    if (c.status != 0)
+        throw new ExitCodeException(c.output);
+
+    return strip(c.output);
+}
+
+
+/***************************************************************************
+
+    Finds out if ref1 is an ancestor of ref2
+
+    Params:
+        ref1 = ref to check if it is an ancestor of ref2
+        ref2 = ref to check if it is a decendant of ref1
+
+    Returns:
+        true if ref1 is an ancestor of ref2
+
+***************************************************************************/
+
+bool isAncestor ( string ref1, string ref2 )
```

Comment:
It wasn't so easy to separate because functions like `getUpstream` interact with the shell and git..